pytest_fauxfactory.py

```diff
@@ -1,135 +1,35 @@
-        item += 1
-
-
-def _pytest_faux_callable_mark_handler(metafunc):
-    """"pytest faux_callable mark handler"""
-    args = metafunc.function.faux_callable.args
-    kwargs = metafunc.function.faux_callable.kwargs
-
-    usage_message = (
-        'usage: faux_callable(items, callable_function, *args, **kwargs)'
+"""Analyse pytest-fauxfactory marks and passes arguments and keywords to
+pytest's parametrize method."""
+from handlers import (
+    callable_mark_handler,
+    generator_mark_handler,
+    string_mark_handler,
     )
 
-    if len(args) < 2:
-        raise pytest.UsageError(
-            'Missing arguments: {0}'.format(usage_message)
-        )
-
-    items, callable_function = args[0:2]
-    if not isinstance(items, int):
-        raise pytest.UsageError(
-            'Mark expected an integer, got a {}: {}'.format(
-                type(items), items))
-    if items < 1:
-        raise pytest.UsageError(
-            'Mark expected an integer greater than 0, got {}'.format(
-                items))
-    if not callable(callable_function):
-        raise pytest.UsageError(
-            'Mark expected a callable function, got a {}: {}'.format(
-                type(callable_function), callable_function))
-
-    return faux_callable(items, callable_function, *args[2:], **kwargs)
-
-
-def _pytest_faux_generator_mark_handler(metafunc):
-    """"pytest faux_generator mark handler."""
-    args = metafunc.function.faux_generator.args
-    usage_message = 'usage: faux_generator(generator)'
-
-    if len(args) == 0:
-        raise pytest.UsageError(
-            'Missing arguments, {0}'.format(usage_message)
-        )
-    for index, arg in enumerate(args):
-        if not isgenerator(arg):
-            raise pytest.UsageError(
-                'Argument with index {0} is not a generator, {1}'
-                .format(index, usage_message)
-            )
-
-    return faux_generator(*args)
-
-
-def _pytest_faux_string_mark_handler(metafunc):
-    """"pytest faux_string mark handler"""
-    # We should have at least the first 2 arguments to faux_string
-    args = metafunc.function.faux_string.args
-    if len(args) == 0:
-        args = (1, None)
-    elif len(args) == 1:
-        if isinstance(args[0], int):
-            args = (args[0], None)
-        elif isinstance(args[0], str):
-            if args[0] in STRING_TYPES:
-                args = (1, args[0])
-            else:
-                raise pytest.UsageError(
-                    'String type {} is not supported.'.format(args[0])
-                )
-    items, str_type = args[0:2]
-
-    if items < 1:
-        raise pytest.UsageError(
-            'Mark expected an integer greater than 0, got {}'.format(
-                items))
-    kwargs = metafunc.function.faux_string.kwargs
+from helpers import (
+    extract_arguments,
+    extract_keyword_arguments,
+    get_mark_function,
+)
 
-    return faux_string(items, str_type, *args[2:], **kwargs)
+FUNCTIONS = {
```

Comment:
Think MARK_HANDLERS name will be more appropriate, what do you think?

Reply:
and perhaps move it to handlers module, that way we can import only this variable .